packages/studio-ui/src/components/CollapsibleSidebar.tsx

```diff
@@ -0,0 +1,53 @@
+import classNames from "classnames";
+import { PropsWithChildren, useCallback, useState } from "react";
+import { ReactComponent as Arrow } from "../icons/chevron-left.svg";
+
+interface CollapsibleSidebarProps {
+  side: "right" | "left";
+}
+export default function CollapsibleSidebar({
+  side,
+  children,
+}: PropsWithChildren<CollapsibleSidebarProps>): JSX.Element {
+  const [isOpen, setIsOpen] = useState(true);
+  const toggleIsOpen = useCallback(() => setIsOpen((isOpen) => !isOpen), []);
+  const arrowRight =
+    (side === "right" && isOpen) || (side === "left" && !isOpen);
+  const containerStyle = classNames("flex", {
+    "w-1/4": isOpen,
+  });
+
+  const childrenStyle = classNames("flex flex-col grow", {
```

Comment:
Could we move `containerStyle` and `childrenStyle` to be optional props? Something like: 

```
interface CollapsibleSidebarProps {
  getContainerStyles?: (isOpen: booelan) => string,
  getContentStyles?: (isOpen: boolean) => string
}
```

We could have defaults within the Component if these props are un-specified. I think having these props will make this Component even more generic and more re-usable for other use-cases.

Reply:
Would you be OK if we wait for a use case to come up for reusing this component?  I'm just a little worried that this might cause confusion since the `hidden` part of the `contentStyle` is integral to the component's function, and I wouldn't want this to be overwritten, I don't think.  I think if we find a use case for it, it'll give us more info on what exactly we might want to allow customization.  What are your thoughts?